### PR TITLE
feat(security): verify that the tolerant ObjectMapper is used

### PR DESCRIPTION
**Pull request by JoergSiebahn**

### CODE REVIEW

File: sda-commons-web-autoconfigure/src/main/java/org/sdase/commons/spring/boot/web/security/validation/CustomObjectMapperAdvice.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ */
+@Component
+@ConditionalOnMissingBean(name = SdaObjectMapperConfiguration.OBJECT_MAPPER_BUILDER_BEAN_NAME)
+public class CustomObjectMapperAdvice {
+
+  private static final String JACKSON_CONFIGURATION_BEAN_CLASS = "Jackson2ObjectMapperBuilder";
+
+  @PostConstruct
+  public void check() {
+    throw new InsecureConfigurationException(
```
Comment (christopher-cudennec) [inline sda-commons-web-autoconfigure/src/main/java/org/sdase/commons/spring/boot/web/security/validation/CustomObjectMapperAdvice.java:33]: Will we use the exception in more cases? At the moment the name sounds really generic but is only used in one very specific case. We could:
* use a better name
* use another `RuntimeException` (`IllegalStateException`?)

File: sda-commons-web-autoconfigure/src/test/java/org/sdase/commons/spring/boot/web/security/test/ContextUtils.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  /**
+   * Creates a spring test context for testing application startup failures. Using `SpringBootTest`
+   * annotation will prevent the tests from starting. Tests that need dependencies from the web
+   * environment must add `@SpringBootTest(webEnvironment = ...)` to the Spring boot application
+   * class because that is how it is internally checked.
+   *
+   * @param clazz The application class annotated with `SpringBootApplication`
+   * @return An assertable application context.
+   */
```
Comment (christopher-cudennec) [inline sda-commons-web-autoconfigure/src/test/java/org/sdase/commons/spring/boot/web/security/test/ContextUtils.java:30]: How are you supposed to test the startup failure? Example would help.

File: sda-commons-web-autoconfigure/src/test/java/org/sdase/commons/spring/boot/web/security/validation/CustomObjectMapperAdviceTest.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import org.sdase.commons.spring.boot.web.jackson.SdaObjectMapperConfiguration;
+import org.sdase.commons.spring.boot.web.security.exception.InsecureConfigurationException;
+import org.sdase.commons.spring.boot.web.security.test.ContextUtils;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.test.context.SpringBootTest;
+
+class CustomObjectMapperAdviceTest {
+  @Test
+  void shouldPreventStartupIfTracingIsEnabled() {
+    Assertions.assertThat(ContextUtils.createTestContext(NoSdaObjectMapperApp.class))
```
Comment (christopher-cudennec) [inline sda-commons-web-autoconfigure/src/test/java/org/sdase/commons/spring/boot/web/security/validation/CustomObjectMapperAdviceTest.java:21]: Static import?

File: sda-commons-web-autoconfigure/src/main/java/org/sdase/commons/spring/boot/web/security/validation/CustomObjectMapperAdvice.java (hunk 1 above)
Comment (JoergSiebahn) [reply sda-commons-web-autoconfigure/src/main/java/org/sdase/commons/spring/boot/web/security/validation/CustomObjectMapperAdvice.java:33]: Yes, there will be more insecure configuration to fail. And even if not, I think „insecure configuration“ is more meaningful than a generic Java exception.

File: sda-commons-web-autoconfigure/src/test/java/org/sdase/commons/spring/boot/web/security/test/ContextUtils.java (hunk 2 above)
Comment (JoergSiebahn) [reply sda-commons-web-autoconfigure/src/test/java/org/sdase/commons/spring/boot/web/security/test/ContextUtils.java]: added

File: sda-commons-web-autoconfigure/src/test/java/org/sdase/commons/spring/boot/web/security/validation/CustomObjectMapperAdviceTest.java (hunk 3 above)
Comment (JoergSiebahn) [reply sda-commons-web-autoconfigure/src/test/java/org/sdase/commons/spring/boot/web/security/validation/CustomObjectMapperAdviceTest.java]: fixed

**Review by christopher-cudennec — APPROVED** [review]

**Pull request merged**